### ISSUE TITLE
SWRResponse data and error are always undefined if it is a context value.

### ISSUE THREAD

**Issue by dexterleng** [vercel/swr#1763]
# Bug report

## Description / Observed Behavior

When I use `SWRResponse` as the context type, I notice the consumer rerenders after the fetcher succeeds, but `data` and `error` are still `undefined`, `isValidating = true`.

I would like to use the `SWRResponse` type directly rather than have to unwrap and rewrap it in my own Response/Result type.

```
const _SomethingContext = createContext<SWRResponse<Something, any> | null>(null);

export const SomethingProvider: FC = (props) => {
  const somethingResponse = useSWR("something", fetchSomething);
  return <_PresaleContext.Provider value={somethingResponse} {...props} />;
};

export function useSomething(): SWRResponse<Something, any> | null {
  const ctx = useContext(_SomethingContext);
  return ctx;
}
```

In some component:
```
const somethingResponse = useSomething();
// Always prints { data: undefined, error: undefined, isValidating: true, ... }
console.log(somethingResponse);
```

## Expected Behavior

I would expect the provider and the consumer to rerender after the request either succeeds or fails.

## Repro Steps / Code Example

Or share your code snippet or a [CodeSandbox](https://codesandbox.io) link is also appreciated!

## Additional Context

SWR version 1.1.2

**Comment by shuding** [comment]
Thanks for creating the reproduction, I’ll take a look! One thing I’m curious is why are you using a context to pass the SWR value, instead of including it directly in the hook?

```js
export function useSomething(): SWRResponse<Something, any> | null {
  return useSWR("something", fetchSomething);
}
```

**Comment by dexterleng** [comment]
@shuding You're right - its totally unnecessary to use context. I've since removed it. This is no longer a relevant issue, so I can close it - although it is strange that this would happen.

**Comment by jskogeby** [comment]
I've stumbled upon the same issue, version 1.0.2 in my case but seem to be persistent when updating to 1.3.0 also. 

The problem only seems to appear after a certain depth in the component tree, as in logging the SWR object in the direct child to the context provider seems to correctly log data and rerender but logging a few children down it remains undefined.

I've also noticed that logging data in the context or in a direct child to the provider has the side-effect that data becomes defined for all children, kind of like an alive Schrödinger's cat scenario.
